src/google/protobuf/arena.h

```diff
@@ -338,8 +338,10 @@ class PROTOBUF_EXPORT alignas(8) Arena final {
   template <typename T>
   PROTOBUF_ALWAYS_INLINE static T* CreateArray(Arena* arena,
                                                size_t num_elements) {
+#ifndef __INTEL_COMPILER
```

Comment:
What goes wrong here with the Intel compiler?

Reply:
With `icpc` the static assert fails for the `google::protobuf::internal::ExtensionSet::KeyValue` type, even though it certainly fits the definition of POD type and trivially construct-able:
```
/home/khq.kitware.com/chuck.atkins/Code/protobuf/source/src/google/protobuf/arena.h(342): error: static assertion failed with "CreateArray requires a trivially constructible type"
      static_assert(std::is_pod<T>::value,
      ^
          detected during instantiation of "T *google::protobuf::Arena::CreateArray<T>(google::protobuf::Arena *, size_t={unsigned long}) [with T=google::protobuf::internal::ExtensionSet::KeyValue]" at line 193 of "/home/khq.kitware.com/chuck.atkins/Code/protobuf/source/src/google/protobuf/extension_set.cc"

compilation aborted for /home/khq.kitware.com/chuck.atkins/Code/protobuf/source/src/google/protobuf/extension_set.cc (code 2)
make[2]: *** [CMakeFiles/libprotobuf.dir/home/khq.kitware.com/chuck.atkins/Code/protobuf/source/src/google/protobuf/extension_set.cc.o] Error 2
make[1]: *** [CMakeFiles/libprotobuf.dir/all] Error 2
make: *** [all] Error 2
```

Reply:
Added comment noting that `icc` mis-evaluates some types as non-pod